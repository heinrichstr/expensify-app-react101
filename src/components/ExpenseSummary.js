import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import getExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

//{numeral(amount / 100).format('$0,0.00')} 

export const ExpenseSummary = (props) => (
    <div>
        <p>Viewing {
            `${props.expenses.length} expense${props.expenses.length > 1 ? 's' : '' }`
        } totalling {
            numeral(getExpensesTotal(props.expenses) / 100).format('$0,0.00')
        }</p>
    </div>
);

const mapStateToProps = (state) => {
    return {
      expenses: selectExpenses(state.expenses)
    };
};
  
export default connect(mapStateToProps)(ExpenseSummary);