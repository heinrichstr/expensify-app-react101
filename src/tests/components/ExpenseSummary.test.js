import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with correct data', () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with single expense', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[2]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with no expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});