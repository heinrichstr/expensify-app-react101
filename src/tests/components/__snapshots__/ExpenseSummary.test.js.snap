// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with correct data 1`] = `
<div>
  <p>
    Viewing 
    3 expenses
     totalling 
    $1,141.95
  </p>
</div>
`;

exports[`should render ExpenseSummary with single expense 1`] = `
<div>
  <p>
    Viewing 
    1 expense
     totalling 
    $45.00
  </p>
</div>
`;
